subprojects/core/src/main/java/org/gradle/caching/internal/tasks/DefaultTaskOutputCachingBuildCacheKeyBuilder.java

```diff
@@ -101,26 +104,33 @@ public void appendInputPropertyHash(String propertyName, HashCode hashCode) {
         hasher.putString(propertyName);
         hasher.putHash(hashCode);
         inputHashes.put(propertyName, hashCode);
-        LOGGER.info("Appending inputPropertyHash for '{}' to build cache key: {}", propertyName, hashCode);
+        log("Appending inputPropertyHash for '{}' to build cache key: {}", propertyName, hashCode);
     }
 
     @Override
     public void inputPropertyLoadedByUnknownClassLoader(String propertyName) {
         valid = false;
         String sanitizedPropertyName = propertyName.replace(".class", "");
         inputPropertiesLoadedByUnknownClassLoader.add(sanitizedPropertyName);
-        LOGGER.info("The implementation of '{}' cannot be determined, because it was loaded by an unknown classloader", sanitizedPropertyName);
+        LogLevel logLevel = buildCacheDebugLogging ? LogLevel.LIFECYCLE : LogLevel.INFO;
+        LOGGER.log(logLevel, "The implementation of '{}' cannot be determined, because it was loaded by an unknown classloader", sanitizedPropertyName);
     }
 
     @Override
     public void appendOutputPropertyName(String propertyName) {
         outputPropertyNames.add(propertyName);
         hasher.putString(propertyName);
-        log("outputPropertyName", propertyName);
+        logAppend("outputPropertyName", propertyName);
     }
 
-    private static void log(String name, @Nullable Object value) {
-        LOGGER.info("Appending {} to build cache key: {}", name, value);
+    private void logAppend(String name, @Nullable Object value) {
+        log("Appending {} to build cache key: {}", name, value);
```

Comment:
Given that this is called a lot I'd inline the call here, so we don't create a ton of `Object[]`s.

Reply:
Another option would be to wrap the non-logging version in a logger decorator, so that without the switch we'd be running at full speed. I think this option would be preferable.

Reply:
I wrapped the logger.